### PR TITLE
Facets: Only show relevant ones on empty product list

### LINKED ISSUE #500: Empty Parts list shows hidden facets like `Worksin` and `Tool Category`
https://www.ifixit.com/Parts/Mac?q=galaxy%20tab

image

Maybe it's not so bad if we need to hide Tool Category and Worksin.

CC: @jeffsnyder

### PR DESCRIPTION
**Pull request by sumukhipandey**
This PR hides `Worksin` on both `/Parts` and `/Tools` and `Tools Category` on `/Parts` when the search query results in an empty product list state.

## QA
- `Tool Category` should now always be hidden on `/Parts` and its children. And on empty state:
   - `Tool Category` and `Worksin` should be hidden on `/Parts`
   - `Worksin` should be hidden on `/Tools`

Closes #500 

CC: @erinemay @jeffsnyder

### CODE REVIEW

**Comment by sumukhipandey** [comment]
This seemed like a straightforward fix to me and since #465 is a tricky one to get I decided to do a separate pull for the empty state.

**Comment by sumukhipandey** [comment]
dev_block 🦋

**Comment by sumukhipandey** [comment]
un_dev_block 🦋

**Comment by deltuh-vee** [comment]
dev_block ✊ 
Tool Category is still appearing for empty Parts pages.

CleanShot 2022-07-27 at 14 43 23

**Comment by sumukhipandey** [comment]
> dev_block ✊ Tool Category is still appearing for empty Parts pages.

Screenshot 2022-07-27 145107

@deltuh-vee I can't seem to see it when I look up something random. Did this happen every time you types random something into the search box?

**Comment by sumukhipandey** [comment]
un_dev_block 🦋

**Comment by deltuh-vee** [comment]
Is Tool Category supposed to show up for pages like `/Parts/iPhone`? This is from the Github preview(not when running react-commerce locally).
CleanShot 2022-07-27 at 15 48 59

**Comment by sumukhipandey** [comment]
> Is Tool Category supposed to show up for pages like `/Parts/iPhone`? This is from the Github preview(not when running react-commerce locally). CleanShot 2022-07-27 at 15 48 59

Yes I believe so because this PR just hides the facets when it hits an empty state. The logic behind displaying facets for different pages is still the same so if it's showing up then it is supposed to because that logic is untouched. Which makes me think, should we hide `Tools Category` when we have an empty state or just leave it grayed out? @sterlinghirsh @masonmcelvain

**Comment by sumukhipandey** [comment]
deploy_block 🦋 on https://github.com/iFixit/react-commerce/pull/515#issuecomment-1197458192

**Comment by sterlinghirsh** [comment]
Don't show tool category on Parts pages

**Comment by sumukhipandey** [comment]
> Thanks for taking this on @sumukhipandey! Also, anything that's done in FacetsAccordian will likely need to be done in FacetsDrawer too, since that is the mobile analog of FacetsAccordian

@masonmcelvain In case of a mobile device, when we have an empty product list displayed there are no facets displayed at all instead of grayed out facets. 
Screenshot 2022-07-28 071826

So this change wouldn't be applicable there correct?

**Comment by masonmcelvain** [comment]
> However, if we have a mobile device and have an empty product list, no facets are displayed at all instead of grayed out facets.

Didn't realize that, I'd say you're right that we don't need to touch anything in `FacetsDrawer` then! Thanks for checking.

**Comment by sumukhipandey** [comment]
un_deploy_block 🦋 
`Tool Category` should now always be hidden on `/Parts` and its children. And on empty state:
- `Tool Category` and `Worksin` should be hidden on `/Parts`
- `Worksin` should be hidden on `/Tools`

File: frontend/components/product-list/sections/FilterableProductsSection/FacetsAccordion.tsx [hunk 1]
```diff
@@ -98,7 +98,24 @@ export const FacetAccordionItem = forwardRef<FacetAccordionItemProps, 'div'>(
       const isDisabled = isProductListEmpty || !hasApplicableRefinements;
 
       const formattedFacetName = formatFacetName(attribute);
-      const isHidden = !hasApplicableRefinements && !isProductListEmpty;
+      let isHidden = !hasApplicableRefinements && !isProductListEmpty;
+      const emptyProductList = isProductListEmpty && !hasApplicableRefinements;
```
Comment (masonmcelvain) [inline frontend/components/product-list/sections/FilterableProductsSection/FacetsAccordion.tsx:102]: Naming: At first glance, it's unclear what the difference between `isProductListEmpty` and `emptyProductList` is. Maybe something like `isUnrefinedEmptyProductList`? That is more verbose tho

File: frontend/components/product-list/sections/FilterableProductsSection/FacetsAccordion.tsx [hunk 2]
```diff
@@ -98,7 +98,24 @@ export const FacetAccordionItem = forwardRef<FacetAccordionItemProps, 'div'>(
...
+
+      let isPartsPage = productList.type === ProductListType.AllParts;
+
+      const ancestors = productList.ancestors;
+      if (ancestors[ancestors.length - 1]) {
+         const path = ancestors[ancestors.length - 1].path;
+         isPartsPage = path.includes('/Parts');
+      }
```
Comment (masonmcelvain) [inline frontend/components/product-list/sections/FilterableProductsSection/FacetsAccordion.tsx:110]: Will this produce the same effect? I'm don't understand the purpose of looking at the path of the second to last ancestor, if I'm missing something lmk.

```suggestion
      const isPartsPage =
         productList.type === ProductListType.AllParts
         || productList.type === ProductListType.DeviceParts;
```

File: frontend/components/product-list/sections/FilterableProductsSection/FacetsAccordion.tsx [hunk 3]
```diff
@@ -98,7 +98,24 @@ export const FacetAccordionItem = forwardRef<FacetAccordionItemProps, 'div'>(
       const isDisabled = isProductListEmpty || !hasApplicableRefinements;
 
       const formattedFacetName = formatFacetName(attribute);
-      const isHidden = !hasApplicableRefinements && !isProductListEmpty;
+      let isHidden = !hasApplicableRefinements && !isProductListEmpty;
```
Comment (masonmcelvain) [inline frontend/components/product-list/sections/FilterableProductsSection/FacetsAccordion.tsx:101]: To keep this boolean immutable, I personally prefer

```suggestion
      const isHidden = !hasApplicableRefinements && !isProductListEmpty && !isWorksin && !isToolsCategoryOnPartsPage;
```

or something like that

**Review by masonmcelvain — APPROVED** [review]
CR ✌🏻 Looks great! A few questions about making the logic a little easier to follow

File: frontend/components/product-list/sections/FilterableProductsSection/FacetsAccordion.tsx (hunk 3 above)
Comment (sumukhipandey) [reply frontend/components/product-list/sections/FilterableProductsSection/FacetsAccordion.tsx]: Resolved 451fdd85a5550cd34a1f611c1383149be57d57ac

File: frontend/components/product-list/sections/FilterableProductsSection/FacetsAccordion.tsx (hunk 2 above)
Comment (sumukhipandey) [reply frontend/components/product-list/sections/FilterableProductsSection/FacetsAccordion.tsx]: Resolved 451fdd85a5550cd34a1f611c1383149be57d57ac

File: frontend/components/product-list/sections/FilterableProductsSection/FacetsAccordion.tsx (hunk 1 above)
Comment (sumukhipandey) [reply frontend/components/product-list/sections/FilterableProductsSection/FacetsAccordion.tsx]: Resolved 451fdd85a5550cd34a1f611c1383149be57d57ac

**Comment by sumukhipandey** [comment]
Thanks @masonmcelvain for all the suggestions! They have been addressed here 451fdd85a5550cd34a1f611c1383149be57d57ac.
As far as naming the variable is concerned, I changed it to `productListEmptyState` because that is what is being checked there. I hope that's clearer.

**Comment by deltuh-vee** [comment]
QA 🎬 
Tools Category is always hidden for `/Parts` and Worksin doesn't appear for `/Parts` or `/Tools`.

**Review by masonmcelvain — APPROVED** [review]
CR ✌🏻 Thanks, I find that much easier to read :)

**Pull request merged**